src/Features/Core/Portable/UnusedReferences/ProjectAssets/ProjectAssetsReader.cs

```diff
@@ -146,23 +138,43 @@ internal static class ProjectAssetsReader
 
                 if (targetLibrary.Compile != null)
                 {
-                    compilationAssemblies.AddRange(targetLibrary.Compile.Keys
-                        .Where(assemblyPath => !assemblyPath.EndsWith(NuGetEmptyFileName))
-                        .Select(assemblyPath => Path.GetFullPath(Path.Combine(packagesPath, library.Path, assemblyPath))));
+                    foreach (var kvp in targetLibrary.Compile)
+                    {
+                        var assemblyPath = kvp.Key;
+                        if (!assemblyPath.EndsWith(NuGetEmptyFileName))
+                        {
+                            compilationAssemblies.Add(Path.GetFullPath(Path.Combine(packagesPath, library.Path ?? "", assemblyPath)));
+                        }
+                    }
                 }
             }
 
             if (referenceType == ReferenceType.Unknown)
             {
                 return null;
             }
+            
+            if (referenceType == ReferenceType.Package && itemSpecification == ".NETStandard.Library")
+            {
+                // This depenedency is large and not useful in determining whether the parent reference has been used.
```

Comment:
i don't really understand the comment.  why is this reference not useful in determining if the parent reference was used?

Reply:
I will update the comment, but the way I am thinking about it is that the .NETStandard.Library brings in the entire BCL as dependencies. So, if I reference Newtonsoft.Json and it references .NETStandard.Library, then it isn't useful that I would consider any use of BCL assemblies as a use of Newtonsoft.Json.